pkg/server/router/credential.go

```diff
@@ -122,6 +123,8 @@ func (cr CredentialRouter) CreateCredential(ctx context.Context, w http.Response
 	}
 
 	resp := CreateCredentialResponse{Credential: createCredentialResponse.Credential, CredentialJWT: createCredentialResponse.CredentialJWT}
+
+	go cr.service.Webhook.PublishWebhook(webhook.Credential, webhook.Create, resp)
```

Comment:
This approach couples the business logic of every endpoint with webhook concerns. 

An alternative that separate these concerns is to have a middleware that is configured per endpoint. Some added benefits include:
* You don't need to update the files in the `router` package.
* Routers do not need to be aware of the `webhook.Service` class.
* It's easy to turn off webhooks via configs.
* More easily manage concurrency (vs spinning a goroutine per request).

Reply:
Sounds good! I will look into the middleware solution.

* More easily manage concurrency (vs spinning a goroutine per request).
On this point ^ I think it will still basically spin up a goroutine per request, but I can add a new context and time it out / maybe do some caching so I don't have to access redis so much

Reply:
Yes, that definitely works. I meant that you could more easily change that model. 

An alternative model is to have every request write to a channel, and have a pool of goroutines read from them and send the outward events.